mesonbuild/compilers/mixins/clike.py

```diff
@@ -347,6 +347,18 @@ def check_header(self, hname: str, prefix: str, env: 'Environment', *,
                      dependencies: T.Optional[T.List['Dependency']] = None) -> T.Tuple[bool, bool]:
         code = f'''{prefix}
         #include <{hname}>'''
+
+        # Change extra_args to list of strings
+        if callable(extra_args):
+            extra_args = extra_args("compile")
+
+        if extra_args is not None:
```

Comment:
This causes the build dir not to be added in all circumstances. Shouldn't the build dir be added unconditionally?

Reply:
In this case the build dir is only added if the include_directories parameter is used and does not already include the build dir.  I thought it would be a more natural usage like this. In addition the include_directories parameter has to be used anyways as far my understanding goes.

Reply:
Well, this currently checks if *any* args are passed (so technically adding `-O2` should also trigger this). Also, it might be better to move this code to https://github.com/mesonbuild/meson/blob/master/mesonbuild/interpreter/compiler.py. This should also give you more options.

Additionally, if adding the build dir is conditional (which I am still not sure that it is a good thing), it also needs to be documented.